src/SdkCommon/ClientRuntime/ClientRuntime/RetryAfterDelegatingHandler.cs

```diff
@@ -44,35 +53,72 @@ public RetryAfterDelegatingHandler(DelegatingHandler innerHandler)
         protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
             CancellationToken cancellationToken)
         {
+            HttpResponseMessage previousResponseMessage = null;
             do {
-                var response = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
-                
-                // if they send back a 429
-                if( response.StatusCode == ((HttpStatusCode)429) ) {
+                HttpResponseMessage response = null;
 
-                    // and there is a retry-after header
-                    if (response.Headers.Contains("Retry-After"))
+                try
+                {
+                    response = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
+
+                    // if they send back a 429
+                    if (response.StatusCode == (HttpStatusCode)429)
```

Comment:
Nit: these 2 if statements could be collapsed to reduce indentation.

Reply:
Was trying to leave the existing code alone as much as possible, but I'm not opposed to cleaning it up some.

Reply:
Ah -- didn't realize this was existing code -- I am fine with not churning that code in this PR.